src/model/release.rs

```diff
@@ -1,14 +1,52 @@
 use serde::Deserialize;
+use std::fmt::{Display, Formatter};
 
 #[derive(Deserialize, Debug)]
 pub struct Release {
     pub tag_name: String,
     pub assets: Vec<Asset>,
 }
 
-#[derive(Deserialize, Debug, Clone)]
+#[derive(Deserialize, Debug, Clone, Eq, PartialEq)]
 pub struct Asset {
     pub id: u32,
     pub name: String,
     pub size: u64,
 }
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum AssetError {
+    /// Asset name of this OS is unknown
+    NameUnknown,
+
+    /// Asset name is not in the fetched assets
+    NotFound(String),
+}
+
+impl Display for AssetError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::NameUnknown => {
+                write!(
+                    f,
+                    "Don't know the asset name for this OS: specify it explicitly in the config"
+                )
```

Comment:
Let's also improve this error message to provide more details for the user (similar to `model/asset_name.rs`)

```suggestion
                write!(
                    f,
                    "Unknown asset selector for OS: {}. Specify 'asset_name.your_os' in the cofig.",
                    env::consts::OS
                )
```

Reply:
good idea